dask_planner/src/sql/types.rs

```diff
@@ -1,107 +1,312 @@
+                                let res: PyResult<String> = e.extract();
+                                Some(res.unwrap())
+                            }
+                            None => None,
+                        };
+                        let unit: TimeUnit = match dict.get_item("unit") {
+                            Some(e) => {
+                                let res: PyResult<&str> = e.extract();
+                                match res.unwrap() {
+                                    "Second" => TimeUnit::Second,
+                                    "Millisecond" => TimeUnit::Millisecond,
+                                    "Microsecond" => TimeUnit::Microsecond,
+                                    "Nanosecond" => TimeUnit::Nanosecond,
+                                    _ => TimeUnit::Nanosecond,
+                                }
+                            }
+                            // Default to Nanosecond which is common if not present
+                            None => TimeUnit::Nanosecond,
+                        };
+                        (unit, tz)
+                    }
+                    // Default to Nanosecond and None for tz which is common if not present
+                    None => (TimeUnit::Nanosecond, None),
+                };
+                DataType::Timestamp(unit, tz)
+            }
+            SqlTypeName::TIMESTAMP => {
+                let (unit, tz) = match py_kwargs {
+                    Some(dict) => {
+                        let tz: Option<String> = match dict.get_item("tz") {
+                            Some(e) => {
+                                let res: PyResult<String> = e.extract();
+                                Some(res.unwrap())
+                            }
+                            None => None,
+                        };
+                        let unit: TimeUnit = match dict.get_item("unit") {
+                            Some(e) => {
+                                let res: PyResult<&str> = e.extract();
+                                match res.unwrap() {
+                                    "Second" => TimeUnit::Second,
+                                    "Millisecond" => TimeUnit::Millisecond,
+                                    "Microsecond" => TimeUnit::Microsecond,
+                                    "Nanosecond" => TimeUnit::Nanosecond,
+                                    _ => TimeUnit::Nanosecond,
+                                }
+                            }
+                            // Default to Nanosecond which is common if not present
+                            None => TimeUnit::Nanosecond,
+                        };
+                        (unit, tz)
+                    }
+                    // Default to Nanosecond and None for tz which is common if not present
+                    None => (TimeUnit::Nanosecond, None),
+                };
+                DataType::Timestamp(unit, tz)
```

Comment:
Do we want to pass the same timezone even when the type is a `timestamp` type without a timezone?

Reply:
I honestly don't know? You would know much better. Which way is the correct way?

Reply:
A default timestamp (without timezone) based on postresql is interpreted as the timezone in utc irrespective of the offset provided. So in this case the tz should always be utc.

Reply:
That might have been handled by the library but either way I added a check just to be sure.